skymp5-server/cpp/server_guest_lib/MpActor.cpp

```diff
@@ -398,17 +409,51 @@ void MpActor::EatItem(uint32_t baseId, espm::Type t)
   } else {
     return;
   }
-
+  std::unordered_set<std::string> modFiles = { GetParent()->espmFiles.begin(),
+                                               GetParent()->espmFiles.end() };
+  bool hasSweetpie = modFiles.count("SweetPie.esp");
```

Comment:
I think this is useful not only for sweetpie. Maybe redo this as optional configuration parameters?

Reply:
using such check is actually acceptable. there is already such check in codebase so it's pretty consistent.

the ideal solution would be to implement this on gamemode side